addons/hr_contract/static/tests/hr_contract_mock_models.js

```diff
@@ -0,0 +1,143 @@
+import { fields, models, defineModels } from "@web/../tests/web_test_helpers";
+import { Domain } from "@web/core/domain";
+import { hrModels } from "@hr/../tests/hr_test_helpers";
+
+export class HrContract extends models.Model {
+    _name = "hr.contract";
+
+    name = fields.Char();
+    state = fields.Selection({
+        selection: [
+            ["draft", "New"],
+            ["open", "Running"],
+            ["close", "Expired"],
+            ["cancel", "Cancelled"],
+        ],
+    });
+    kanban_state = fields.Selection({
+        selection: [
+            ["normal", "Grey"],
+            ["done", "Green"],
+            ["blocked", "Red"],
+        ],
+    });
+    employee_id = fields.Many2one({ relation: "hr.employee" });
+    date_start = fields.Date();
+    date_end = fields.Date();
+
+    _records = [
+        {
+            id: 1,
+            name: "Contract - Pig",
+            employee_id: 1,
+            kanban_state: false,
+            state: false,
+        },
+    ];
+
+    get_employee_working_periods(employees, start, stop) {
+        const rows = {};
+
+        for (const val of employees) {
+            rows[val] = { working_periods: [] };
+        }
+
+        const employees_with_contract = this.env["hr.contract"].formatted_read_group(
+            new Domain([
+                "&",
+                ["employee_id", "in", employees],
+                "|",
+                ["state", "not in", ["draft", "cancel"]],
+                "&",
+                ["kanban_state", "=", "done"],
+                ["state", "=", "draft"],
+            ]).toList(),
```

Comment:
You can just give the domain list, I don't understand why you need to create domain object to convert it in domain list. 🤔